lib/ansiblelint/formatters/__init__.py

```diff
@@ -41,69 +43,72 @@ def _format_path(self, path):
-                                    colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber),
-                                    colorize(u"{0}".format(match.details), Color.line))
+            return formatstr.format(
+                colorize(u"[{0}]".format(_id), Color.error_code),
+                colorize(match.message, Color.error_title),
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber),
+                colorize(u"{0}".format(match.details), Color.line))
         else:
             return formatstr.format(_id,
                                     match.message,
-                                    match.filename,
+                                    match.filename or "",
                                     match.linenumber,
                                     match.details)
 
 
 class QuietFormatter(BaseFormatter):
 
-    def format(self, match, colored=False):
+    def format(self, match: MatchError, colored: bool = False) -> str:
         formatstr = u"{0} {1}:{2}"
         if colored:
-            return formatstr.format(colorize(u"[{0}]".format(match.rule.id), Color.error_code),
-                                    colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber))
+            return formatstr.format(
+                colorize(u"[{0}]".format(match.rule.id), Color.error_code),
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber))
         else:
-            return formatstr.format(match.rule.id, self.format_path(match.filename),
+            return formatstr.format(match.rule.id, self._format_path(match.filename or ""),
                                     match.linenumber)
 
 
 class ParseableFormatter(BaseFormatter):
 
-    def format(self, match, colored=False):
+    def format(self, match: MatchError, colored: bool = False) -> str:
         formatstr = u"{0}:{1}: [{2}] {3}"
         if colored:
-            return formatstr.format(colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber),
-                                    colorize(u"E{0}".format(match.rule.id), Color.error_code),
-                                    colorize(u"{0}".format(match.message), Color.error_title))
+            return formatstr.format(
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber),
+                colorize(u"E{0}".format(match.rule.id), Color.error_code),
+                colorize(u"{0}".format(match.message), Color.error_title))
```

Comment:
unrelated?

Reply:
ditto, not trying to slip anything not required.